### PR TITLE
Move plugin config to root.pipeline.[step].settings 

### PR DESCRIPTION
**Pull request by jolheiser**
This PR moves vargs to settings.

More specifically, it moves

```yaml
pipline:
  something:
    image: my/plugin
    setting1: foo
    setting2: bar
```
to
```yaml
pipline:
  something:
    image: my/plugin
    settings:
      setting1: foo
      setting2: bar
```

-----

1. Should we attempt to offer some form of migratory tool? With a copy of our container before/after structs we could _possibly_ move `Vargs` to nested `Settings`
2. Do we want to implement getting the settings from secrets right away?

### CODE REVIEW

**Comment by 6543** [comment]
Idear is if version is not set it's v1

And your breaking changes is v2

Let tje v1 parse stay as is and ad a new one?
reuse code where posible.

**Comment by 6543** [comment]
To prefent duplicated code if posible iproposed a common package

File: pipeline/frontend/yaml/container.go [hunk 1]
```diff
@@ -83,5 +89,15 @@ func (c *Containers) UnmarshalYAML(value *yaml.Node) error {
 			c.Containers = append(c.Containers, &container)
 		}
 	}
+
+	// FIXME deprecate Vargs -> Settings
```
Comment (6543) [inline pipeline/frontend/yaml/container.go:93]:
```suggestion
	// TODO: drop Vargs in favour of Settings in v1.16.0 release
```

**Comment by anbraten** [comment]
TODO note: update json schema

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
@anbraten can you add the schema part to this pull?

**Comment by anbraten** [comment]
@6543 I updated the schema a bit and fixed tests.

**Review by 6543 — APPROVED** [review]

**Pull request merged**